pylon_camera_parameters:
  exposure:
    type: double
    default_value: 3000.0
    description: "The exposure time in microseconds to be set after opening the camera."
    validation:
      bounds<>: [0.0, 1000000.0]
  gain:
    type: int
    default_value: 200
    description: "The target raw gain of the camera sensor (similar to ISO)."
    validation:
      bounds<>: [0, 360]
  fps:
    type: double
    default_value: 10.0
    read_only: true
    description: "Target frame rate of the camera / publisher."
    validation:
      bounds<>: [0.0, 30.0]
  binning_factor_x:
    type: int
    default_value: 4
    read_only: true
    description: "Binning factor to get downsampled images in x direction."
    validation:
      gt_eq<>: [1]
  binning_factor_y:
    type: int
    default_value: 4
    read_only: true
    description: "Binning factor to get downsampled images in y direction."
    validation:
      gt_eq<>: [1]
  camera_info_url:
    type: string
    default_value: ""
    read_only: true
    description: "The URL of the camera calibration file."
    validation:
      not_empty<>: []
  device_user_id:
    type: string
    default_value: ""
    read_only: true
    description: "The name of the camera (used to discover the camera). The name can be set in the pylon viewer."
    validation:
      not_empty<>: []
  camera_frame_id:
    type: string
    default_value: "camera_optical_frame"
    read_only: true
    description: "The tf frame at which the camera's optical center is located."
    validation:
      not_empty<>: []
  gige:
    mtu_size:
      type: int
      default_value: 9000
      description: "The MTU size for GigE cameras."
      read_only: true

    inter_pkg_delay:
      type: int
      default_value: 1000
      description: "The inter-package delay in 'ticks' for GigE cameras."
      read_only: true
      validation:
        bounds<>: [0, 10000]
